app/views/govuk_publishing_components/components/_summary_list.html.erb

```diff
@@ -68,16 +66,25 @@
     <% if items.any? %>
       <%= tag.dl class: "govuk-summary-list" do %>
         <% items.each do |item| %>
-          <%= tag.div class: "govuk-summary-list__row" do %>
+          <%
+            title_actions =  (edit_main_link || delete_main_link)
+            body_actions = (item.fetch(:edit, {}).any? || item.fetch(:delete, {}).any?)
+
+            if !title_actions || (title_actions && body_actions) 
+              classes = "govuk-summary-list__row"
+            else title_actions
+              classes = "govuk-summary-list__row govuk-summary-list__row--no-actions"
+            end
+              classes
+          %>
+          <%= tag.div class: classes do %>
 
```

Comment:
I have spotted an issue with the way the `govuk-summary-list__row--no-actions` class is added when we have title actions (but no "edit" or "delete" action for an individual item). Which creates additional space.

![govuk-publishing-components dev gov uk_component-guide_summary_list](https://user-images.githubusercontent.com/87758239/156396339-08c7b0a2-8b1a-4834-8c7d-c50d2d414e67.png)

So, I think we need to check for the existence of any occurences of "edit" or "delete" actions in the entire array and then use that in the loop when checking if we have a mix of rows with and without actions?

Perhaps something like this: 

```
<% any_body_actions = items.detect { |item| item.has_key?("edit") || item.has_key?("delete") } %>

<% if items.any? %>
  <%= tag.dl class: "govuk-summary-list" do %>
    <% items.each do |item| %>
      <%
        item_body_actions = (item.fetch(:edit, {}).any? || item.fetch(:delete, {}).any?)

        if any_body_actions.present? && !item_body_actions.present?
          classes = "govuk-summary-list__row govuk-summary-list__row--no-actions"
        else
          classes = "govuk-summary-list__row"
        end
      %>
      <%= tag.div class: classes do %>
```

Reply:
> Which creates additional space.

So, this is intentional - from my understanding of this mixed state of actions at the top and no actions, we want to create this faux column, the `no-actions` creates this space. It's sort of the reverse of the below:

![image](https://user-images.githubusercontent.com/71266765/156399877-89d57fbc-188a-4cc5-915b-75d7b92101ed.png)

Reply:
I could be wrong (it's tricky because there is no equivalent in the design system for title actions).

But from the way I read the other examples, it seems we only want to add the extra column when you have a mixture of rows with and without actions, so that the columns align correctly.

Title actions, which appear aligned to the title, and not in the definition list, don't affect the alignment of the columns so we can stretch the description to the right margin and no need for the extra column.

I see what you mean though and it could be intentional.

Reply:
Questioning my whole comprehension of this issue.  So my understanding is that was the original issue https://github.com/alphagov/govuk-frontend/issues/2495

![image](https://user-images.githubusercontent.com/71266765/156416504-899cbe1f-cf64-42ed-9dd0-9dad6f4d782d.png)

So `--no-actions` was added otherwise if it's removed this 'extra space" is still present

https://user-images.githubusercontent.com/71266765/156416565-2fdb325d-c5f3-45df-bebd-c27dfd308b46.mov

> (it's tricky because there is no equivalent in the design system for title actions).

Agreed, whether this extra space should even exist on the gem version _if_ the title actions are present i'm not sure.

>when you have a mixture of rows with and without actions, so that the columns align correctly.

I've taken this to mean, visually at least, these are the same - the user doesn't differentiate between the two they see actions and scan vertically so this _should_ apply.

Reply:
> So --no-actions was added otherwise if it's removed this 'extra space" is still present

One thing to add. I discovered that `--no-actions` needs to be removed from every row in order for the columns to stretch to fill the container (as it creates an empty `display: table-cell;` and pretty much a new column for the whole definition list).

https://user-images.githubusercontent.com/87758239/156551592-38264303-2a87-4936-b4f4-c8f1a325fb26.mov

Reply:
This makes sense - as if it's not applied then it would be a "no action" summary list so would extend.  This is a new change that's part of this PR.

![image](https://user-images.githubusercontent.com/71266765/156568610-4278bfa2-51d2-4e0e-b255-6fbbb2ef8852.png)

This modifier persists the faux column when there are mixed actions - in this context where `gem-c-summary-list__group-title` exists.  I think right now my question is more is this understanding correct?  

As it's only on the gem we have `gem-c-summary-list__group-title`.  From a UX POV I believe this is correct but I've asked the DS to confirm either way.

![image](https://user-images.githubusercontent.com/71266765/156569313-fb412003-5f3e-4d7f-b009-d7aaec888dba.png)

Additional action point, there's not an example of a summary-list with no group title _and_ mixed actions. I'll add this.

Reply:
@jon-kirwan after a chat with the DS and reviewing the original issue - I've totally changed my mind.

- Updated with your suggestions
- Extending the full width when there is no action 
- This also applies regardless of the title actions (letting the content flow the full-width)
- Updated the docs to include include this mix